colossalai/booster/plugin/hybrid_parallel_plugin.py

```diff
@@ -1166,18 +1177,28 @@ def configure(
                     pp_process_group=self.pp_group,
                     verbose=True,
                     clip_grad_norm=self.max_norm,
-                    **self.zero_config,
+                    **zero_config,
                     **self.amp_config,
                 )
 
             # Setup optimizers that require global states
-            if isinstance(optimizer.optim, DistributedOptim):
-                tp_group = self.tp_group
-                dp_group = self.dp_group
+            optim = optimizer.optim
+            states_sharded = self.tp_size > 1 or (self.dp_size > 1 and zero_stage > 0)
+            if isinstance(optim, DistributedOptim):
                 shard_to_param = optimizer.get_master_to_working_map()
-                optimizer.optim.setup_distributed(tp_group, dp_group, shard_to_param, is_zero)
+                if isinstance(optim, DistGaloreAwamW8bit):
+                    padding_map = optimizer.get_param_padding_map() if is_zero else None
+                    optim.setup_distributed(self.tp_group, self.dp_group, shard_to_param, padding_map, is_zero)
+                else:
+                    optim.setup_distributed(self.tp_group, self.dp_group, shard_to_param, is_zero)
```

Comment:
They should have the same interface. Add padding_map as an arg for other distributed optim